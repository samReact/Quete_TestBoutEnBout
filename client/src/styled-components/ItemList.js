import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const List = styled.ul`
  margin: 128px auto;
  display: grid;
  width: 1024px;
  grid-template-columns: 256px 256px 256px;
  column-gap: 128px;
  row-gap: 128px;
  padding-inline-start: 0;
`;

export const ListElement = styled.li`
  display: block;
`;

export const LinkToDetails = styled(Link)`
  color: initial;
  text-decoration: initial;
`;
