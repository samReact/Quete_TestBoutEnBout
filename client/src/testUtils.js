// eslint-disable-next-line import/prefer-default-export
export const items = [
  {
    id: '1234',
    shortName: 'Glouglou',
    inflatedPrice: 100,
    largestImageUrl: 'http://no.big/deal',
  },
  {
    id: '5678',
    shortName: 'mdr',
    largestImageUrl: 'http://okfine.com/lol',
  },
];
