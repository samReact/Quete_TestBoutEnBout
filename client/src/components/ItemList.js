import React from 'react';
import PropTypes from 'prop-types';
import { Name } from '../styled-components/Item';
import {
  List,
  ListElement,
  LinkToDetails,
} from '../styled-components/ItemList';
import ItemImage from './ItemImage';
import ItemPrice from './ItemPrice';

const ItemList = ({ items }) => (
  <List data-selector="item-list">
    {items.map(item => (
      <ListElement data-selector="item-list-element" key={item.id}>
        <LinkToDetails to={`/items/${item.id}`}>
          <ItemImage imageUrl={item.largestImageUrl} />
          <Name>{item.shortName.toUpperCase()}</Name>
          <ItemPrice value={item.inflatedPrice} />
        </LinkToDetails>
      </ListElement>
    ))}
  </List>
);

ItemList.propTypes = {
  items: PropTypes.array,
};

ItemList.defaultProps = {
  items: [],
};

export default ItemList;
