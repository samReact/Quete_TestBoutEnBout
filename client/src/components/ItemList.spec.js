import React from 'react';
import { shallow } from 'enzyme';
import { items } from '../testUtils';
import ItemList from './ItemList';

describe('ItemList', () => {
  describe('when passed no items', () => {
    it('renders empty list', () => {
      const wrapper = shallow(<ItemList />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when passed items', () => {
    it('should display items', async () => {
      await expect(page).toMatchElement(
        'ul[data-selector=item-list] li[data-selector=item-list-element]'
      );
    });
  });
});
