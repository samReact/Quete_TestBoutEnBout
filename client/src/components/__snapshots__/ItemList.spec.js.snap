// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList when passed items renders list 1`] = `
<styled.ul>
  <styled.li
    data-selector="item-list-element"
    key="1234"
  >
    <Styled(Link)
      to="/items/1234"
    >
      <ItemImage
        imageUrl="http://no.big/deal"
      />
      <styled.div>
        GLOUGLOU
      </styled.div>
      <ItemPrice
        render={[Function]}
        value={100}
      />
    </Styled(Link)>
  </styled.li>
  <styled.li
    data-selector="item-list-element"
    key="5678"
  >
    <Styled(Link)
      to="/items/5678"
    >
      <ItemImage
        imageUrl="http://okfine.com/lol"
      />
      <styled.div>
        MDR
      </styled.div>
      <ItemPrice
        render={[Function]}
        value={null}
      />
    </Styled(Link)>
  </styled.li>
</styled.ul>
`;

exports[`ItemList when passed no items renders empty list 1`] = `<styled.ul />`;
